base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1601,9 +1608,41 @@ fn handle_reorg<T: BlockchainBackend>(
             Box::new(removed_blocks),
             Box::new(added_blocks),
         )))
+    } else {
+        trace!(
+            target: LOG_TARGET,
+            "No reorg required. Num blocks to remove: {}, to add: {}.",
+            num_removed_blocks,
+            num_added_blocks,
+        );
+        Ok(BlockAddResult::Ok)
     }
 }
 
+fn reorg_required(num_removed_blocks: usize, num_added_blocks: usize) -> bool {
+    // if any blocks are being removed
+    // or more than one block is being added
+    // then a reorg is required
+    num_removed_blocks > 0 || num_added_blocks > 1
+}
+
+#[test]
+fn test_reorg_required() {
+    // no blocks removed, only one block added - no reorg
+    assert!(reorg_required(0, 1) == false);
+
+    // more than one block removed - should reorg
+    assert!(reorg_required(1, 0) == true);
```

Comment:
the `==true` is superfluous
```suggestion
    assert!(reorg_required(1, 0));
```

Reply:
I knew someone would catch this :) Will revert